### PR TITLE
docs: Generating sphinx datastructure tables on the fly

### PR DESCRIPTION
**Pull request by cssherman**
This PR allows us to build documentation datastructure tables on the fly, and avoids committing a bunch of generated files to the repo.

### CODE REVIEW

**Review by untereiner — APPROVED** [review]

**Review by paveltomin — APPROVED** [review]

**Comment by rrsettgast** [comment]
@cssherman It looks like #3343 has done its job. This PR now has some readthedocs errors/warnings that are triggering a CI failure.

**Comment by cssherman** [comment]
> @cssherman It looks like #3343 has done its job. This PR now has some readthedocs errors/warnings that are triggering a CI failure.

Sure, I'll take a look!

**Review by victorapm — APPROVED** [review]

**Comment by untereiner** [comment]
I suggest to also keep out from the code repository all restructured text documentation transformed by sphinx (tutorials and so on).
The main pros I see is:
- separation of concerns: code repository for code, doc repository for documentation
- redirection of a number of PRs: focus on the code (and code documentation) in the code repository 
- online edition (with github) more friendly in a smaller and focused repository

It is possible to generate the rst files from the code and put them in an archive as a CI artifact that can be reused in another repository

The main con I foresee now is a more complex CI.

**Comment by cssherman** [comment]
@rrsettgast - this should be ready to go!

File: .readthedocs.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ build:
     nodejs: "16"
     python: "3.10"
   jobs:
+    pre_build:
+    - python3 scripts/SchemaToRSTDocumentation.py -s src/coreComponents/schema/schema.xsd -o src/docs/sphinx
```
Comment (rrsettgast) [inline .readthedocs.yml:19]: @bmhan12 Could we execute our doxygen build here instead of the `conf.py file`?

**Review by rrsettgast — APPROVED** [review]

**Pull request merged**